pkg/storage/iterators.go

```diff
@@ -411,3 +415,146 @@ func NewConditionsFilteredTupleKeyIterator(iter TupleKeyIterator, filter TupleKe
 func IterIsDoneOrCancelled(err error) bool {
 	return errors.Is(err, ErrIteratorDone) || errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded)
 }
+
+type OrderedCombinedIterator struct {
+	mu          *sync.Mutex
+	once        *sync.Once
+	mapper      TupleMapper
+	pending     []TupleIterator  // GUARDED_BY(mu)
+	lastHead    *openfgav1.Tuple // GUARDED_BY(mu)
+	lastYielded *openfgav1.Tuple // GUARDED_BY(mu)
+}
+
+var _ TupleIterator = (*OrderedCombinedIterator)(nil)
+
+// TODO(miparnisari): this interface & impls belong, conceptually, in internal/graph/tuplemapper.go, but i don't want to have storage import graph.
```

Comment:
Where is `graph` being used here?

These already exist, why are we re-declaring them?

Reply:
Definitely

Reply:
https://github.com/openfga/openfga/pull/2226